netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -406,11 +430,40 @@ protected ClientTransportFactory buildTransportFactory() {
       }
       negotiator = createProtocolNegotiatorByType(negotiationType, localSslContext);
     }
+
+    boolean usingSharedGroup = eventLoopGroup == null;
+
+    populateDefaultChannelAndGroup();
+
     return new NettyTransportFactory(
         negotiator, channelFactory, channelOptions,
         eventLoopGroup, flowControlWindow, maxInboundMessageSize(),
         maxHeaderListSize, keepAliveTimeNanos, keepAliveTimeoutNanos, keepAliveWithoutCalls,
-        transportTracerFactory.create(), localSocketPicker);
+        transportTracerFactory.create(), localSocketPicker, usingSharedGroup);
+  }
+
+  private void populateDefaultChannelAndGroup() {
+    boolean groupOrChannelProvided = channelFactory != DEFAULT_CHANNEL_FACTORY
+        || eventLoopGroup != null;
+
+    if (groupOrChannelProvided) {
+      logger.log(
+          Level.WARNING,
+          "Both EventLoopGroup and ChannelType should be provided, otherwise client may not start "
+              + "depends on the gRPC default value.");
```

Comment:
The sentence is broken between "start" and "depends".

Include in the message that we are falling back to Nio for compatibility, and that this will cause an exception in the future.